vminitd/Sources/vminitd/TerminalIO.swift

```diff
@@ -139,15 +138,16 @@ final class TerminalIO: ManagedProcess.IO & Sendable {
         }
     }
 
-    func cleanup(_ fd: Int32, buffer: UnsafeMutableBufferPointer<UInt8>, log: Logger?) {
+    func cleanupRelay(readFd: Int32, writeFd: Int32, buffer: UnsafeMutableBufferPointer<UInt8>, log: Logger?) {
         do {
             // We could alternatively just allocate buffers in the constructor, and free them
             // on close().
             buffer.deallocate()
-            try ProcessSupervisor.default.poller.delete(fd)
+            try ProcessSupervisor.default.poller.delete(readFd)
         } catch {
-            self.log?.error("failed to delete pipe fd from epoll \(fd): \(error)")
+            self.log?.error("failed to delete pipe fd from epoll \(readFd): \(error)")
         }
+        Foundation.close(writeFd)
```

Comment:
nit: Check if < 0 and log the errno. These io issues can get gnarly to look into

Reply:
Same for StandardIO.swift